ERCS/erc-7644.md

```diff
@@ -0,0 +1,200 @@
+---
+eip: 7644
+title: ERC-721 Name Registry Extension
+description: Add a time-limited unique naming registry mechanism to ERC-721 tokens.
+author: Chen Liaoyuan (@chenly)
+discussions-to: https://ethereum-magicians.org/t/erc-7644-erc-721-name-registry-extension/19022
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-03-01
+requires: 721
+---
+
+## Abstract
+
+This extension defines an interface that adds a naming mechanism to [ERC-721](./eip-721.md) tokens. It allows each token to have a unique name with a set expiration date. The interface includes functions for assigning, updating, and querying names and their associated tokens, ensuring that names remain unique until they expire.
```

Comment:
I'd like to see some text explaining the scope and purpose of the names. For example:

- are the names globally unique or just for the specific token contract?
- who sets the names, and who is expected to use the names?